### PR TITLE
Add blinking cutouts!

### PR DESCRIPTION
**Pull request by moeyensj**
Main changes:
- Adds `select_comparison_cutout` which given an RA, Dec, and a observation candidate will select a suitable comparison candidate for "blinking"
- Adds blinking cutouts! (set `compare = True` in `run_cutouts_from_precovery`)
- RA, Dec, and obscode have been added to the plot titles
- Reworked plotting code to use matplotlib axes transformations made available through the WCS 
- Dynamic resizing of images with matplotlib has been turned off so that each frame in a blinking cutout is the same size (required by v3 of imageio). 

Little more detail:
- `add_crosshair` now uses the on-sky coordinates and will align along the RA, Dec axes (so for equatorial mounts it will look like it did before unless the image was flipped but for alt-az mounts it will rotate to indicate the direction of north)
- `add_velocity_vector` now also uses only on-sky coordinates and will scale with the velocity vector and exposure time (with a default scale factor of 10 — if the exposure was 10 times as long the length of the vector is how much it would have moved)
- `center_image` has been removed in favor of using a cool astropy utility that pads images to the desired size and updates the WCS accordingly (meaning we no longer have to track x_offset, y_offset at all)

Backend IO Bug fixes:
- SkyMapper was returning two results per image: one for pngs and one for fits files. Results have been limited to fits files. 
- ZTF prepends a "z" to its filter names in the cutout service response. These have been removed.

### CODE REVIEW

**Comment by moeyensj** [comment]
@ntellis @KatKiker I still need to test this for ZTF and a few more objects, but take a look at `plot.py` and see what you think. Hopefully, this solves the x_offset, y_offset, pixel - world space headaches.

File: cutouts/io/skymapper.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ def find_cutouts_skymapper(
         cutout_request.width_arcsec,
     )
     results = pd.DataFrame(results)
+    # Limit results to just fits files
+    results = results[results["format"] == "image/fits"]
```
Comment (ntellis) [inline cutouts/io/skymapper.py:46]: Good fix, we were just lucking out that the png and fits rows returned both had a "get_fits" parameter.

File: cutouts/plot.py [hunk 2]
```diff
@@ -618,16 +435,39 @@ def plot_cutouts(
...
+            center = SkyCoord(ra_i, dec_i, unit="deg", frame="icrs")
+            image_centered = Cutout2D(
+                hdu.data,
+                center,
+                (cutout_height_arcsec * u.arcsec, cutout_width_arcsec * u.arcsec),
+                wcs=wcs,
+                mode="partial",
+                fill_value=np.nan,
+            )
+            wcs = image_centered.wcs
```
Comment (ntellis) [inline cutouts/plot.py:457]: This is very clean. Nice addition!

File: cutouts/plot.py [hunk 3]
```diff
@@ -111,239 +146,66 @@ def add_velocity_vector(
...
-    y_center = y_center + y_offset
+    # Calculate the unit vector in the direction of the velocity
+    vra_hat = vra / np.sqrt(vra**2 + vdec**2)
+    vdec_hat = vdec / np.sqrt(vra**2 + vdec**2)
 
-    image_width_pixels, image_height_pixels = wcs.array_shape
+    # Convert to the correct units
+    gap_degree = gap / 3600
+    dra = vra * (dt / 86400) * scale_factor
+    ddec = vdec * (dt / 86400) * scale_factor
```
Comment (ntellis) [inline cutouts/plot.py:167]: Do we want to floor these so they don't go off the image plane?

**Review by ntellis — APPROVED** [review]
These are great additions, and clean up plot.py while adding useful functionality.

File: cutouts/plot.py (hunk 3 above)
Comment (moeyensj) [reply cutouts/plot.py:170]: I've modified it so if the velocity vector is going to exceed the image bounds it will be clipped and set to be red.

**Comment by moeyensj** [comment]
@KatKiker @ntellis I extended the scope to include some mock queries we can build off of in the future. Let me know what you think. If you are both still happy, I'll merge the PR.

File: cutouts/io/skymapper.py [hunk 4]
```diff
@@ -82,7 +86,7 @@ def find_cutouts_skymapper(
     return results
 
 
-class Skymapper_SIA(SIAHandler):
+class SKYMAPPER_DR2_SIA(SIAHandler):
```
Comment (akoumjian) [inline cutouts/io/skymapper.py:89]:
```suggestion
class SkyMapper_DR2_SIA(SIAHandler):
```

File: cutouts/io/skymapper.py [hunk 5]
```diff
@@ -34,7 +35,7 @@ def find_cutouts_skymapper(
         f"Fetching Skymapper cutouts with ra: {cutout_request.ra_deg} dec: {cutout_request.dec_deg}."
     )
 
-    sia = Skymapper_SIA()
+    sia = SKYMAPPER_DR2_SIA()
```
Comment (akoumjian) [inline cutouts/io/skymapper.py:38]:
```suggestion
    sia = SkyMapper_DR2_SIA()
```

File: cutouts/io/tests/test_skymapper_dr2.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import os
+from contextlib import contextmanager
+from unittest.mock import patch
+
+from astropy.io.votable import parse
+from pyvo.dal.sia import SIAResults
+
+from ..skymapper import SKYMAPPER_DR2_SIA, find_cutouts_skymapper_dr2
```
Comment (akoumjian) [inline cutouts/io/tests/test_skymapper_dr2.py:8]:
```suggestion
from ..skymapper import SkyMapper_DR2_SIA, find_cutouts_skymapper_dr2
```

File: cutouts/io/tests/test_skymapper_dr2.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    width_arcsec=20.0,
+    delta_time=1e-05,
+)
+
+TEST_REQUESTS = [cutout_request1]
+
+
+@contextmanager
+def mock_sia_skymapper_dr2_query(table_file: str):
+    with patch.object(SKYMAPPER_DR2_SIA, "search") as mock_query:
```
Comment (akoumjian) [inline cutouts/io/tests/test_skymapper_dr2.py:32]:
```suggestion
    with patch.object(SkyMapper_DR2_SIA, "search") as mock_query:
```

File: cutouts/io/tests/testdata/nsc_dr2/README.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+```python
+from pyvo.dal.sia import SIAService
+from cutouts.io.types import CutoutRequest
+from cutouts.io.tests.test_nsc_dr2 import TEST_REQUESTS
+
+SIA_URL = "https://datalab.noirlab.edu/sia/nsc_dr2"
+service = SIAService(SIA_URL)
+
+for cutout_request in cutout_requests:
```
Comment (akoumjian) [inline cutouts/io/tests/testdata/nsc_dr2/README.md:13]:
```suggestion
for cutout_request in TEST_REQUESTS:
```

File: cutouts/main.py [hunk 9]
```diff
@@ -32,13 +33,20 @@ def get_cutouts(
     out_dir: str = "~/.cutouts",
     timeout: Optional[int] = 180,
     use_cache: bool = True,
-) -> Iterable[Dict[str, Any]]:
+    compare: bool = False,
+    compare_kwargs: dict = {
```
Comment (akoumjian) [inline cutouts/main.py:37]: Let's change the default to `None` and add the default within the function call.

File: cutouts/main.py [hunk 10]
```diff
@@ -82,7 +106,44 @@ def get_cutouts(
...
+
+                    try:
+                        download_cutout(
+                            result["cutout_url"],
+                            out_file=path.as_posix(),
+                            cache=True,
+                            pkgname="cutouts",
+                            timeout=timeout,
+                        )
+                    except FileNotFoundError as e:
```
Comment (akoumjian) [inline cutouts/main.py:143]: Let's replace these with generic exception handling, since the behavior should always be the same.

**Comment by moeyensj** [comment]
Generates images like: 
2022PC_abstract

**Pull request merged**